client/dfclient/src/gameplay_state.cpp

```diff
@@ -416,48 +449,35 @@ void GameplayState::ProcessWorldState(const void* ev) {
 	}
 	this->currentHidingSpot = worldState->currentHidingSpot()->str();
 
-	// handle ink particles
-	for (auto& i : this->inkParticles)
-		i.second.seen = false;
-
 	for (int i = 0; i < worldState->inkParticles()->size(); i++) {
-		auto ink = worldState->inkParticles()->Get(i);
-		auto inkItr = inkParticles.find(ink->inkID());
-		if (inkItr == inkParticles.end()) {
-			// not found, make a new one
+		auto inkParticleData = worldState->inkParticles()->Get(i);
+		processMovable(this->inkParticles, inkParticleData->movable(), [&](){
 			InkParticle newInk;
 			int inkNum = rand() % 3 + 1;
 			std::string inkTexName = std::string("ink") + std::to_string(inkNum) + ".png";
 			newInk.sprite = std::make_unique<ncine::Sprite>(this->cameraNode.get(), _resources.textures[inkTexName].get());
 			newInk.sprite->setLayer((unsigned short) Layers::INK_PARTICLES);
 			newInk.sprite->setScale(120./330.); // todo: fix my life
-
-			inkParticles.insert({ink->inkID(), std::move(newInk)});
-			inkItr = inkParticles.find(ink->inkID());
-			inkItr->second.lerp.bind(inkItr->second.sprite.get());
-		}
-		inkItr->second.lerp.setupLerp(ink->pos()->x(), ink->pos()->y(), inkItr->second.sprite->rotation());
-		inkItr->second.seen = true;
+			return std::move(newInk);
+		});
 	}
 
-	for (auto it = this->inkParticles.begin(); it != this->inkParticles.end();) {
-		if (!it->second.seen)
-			it = this->inkParticles.erase(it);
-		else
-			++it;
-	}
+	deleteUnusedMovables(this->inkParticles);
 
-	// handle mob manipulators
-	this->manipulators.clear();
 	for (int i = 0; i < worldState->mobManipulators()->size(); i++) {
-		auto manipulator = worldState->mobManipulators()->Get(i);
-		auto manipTexture = manipulator->type() == FlatBuffGenerated::MobManipulatorType_Dispersor ?
-			"dispersor.png" : "attractor.png";
-		auto sprite = std::make_unique<ncine::Sprite>(this->cameraNode.get(), _resources.textures[manipTexture].get());
-		sprite->setLayer((unsigned short) Layers::MOB_MANIPULATORS);
-		sprite->setPosition({manipulator->pos()->x() * METERS2PIXELS, -manipulator->pos()->y() * METERS2PIXELS});
-		this->manipulators.push_back(std::move(sprite));
-	}
+		auto manipulatorData = worldState->mobManipulators()->Get(i);
+		processMovable(this->manipulators, manipulatorData->movable(), [&](){
+			auto manipTexture = manipulatorData->type() == FlatBuffGenerated::MobManipulatorType_Dispersor ?
+				"dispersor.png" : "attractor.png";
+			auto sprite = std::make_unique<ncine::Sprite>(this->cameraNode.get(), _resources.textures[manipTexture].get());
+			sprite->setLayer((unsigned short) Layers::MOB_MANIPULATORS);
+			Manipulator ret;
+			ret.sprite = std::move(sprite);
+			return std::move(ret);
```

Comment:
`std::move` is not needed when returning, and actually worsens performance. Please just do `return ret;`

This is called a [pessimizing move](https://stackoverflow.com/questions/62061433/how-to-avoid-the-pessimizing-move-warning-of-nrvo).

Reply:
Removed `std::move`s. Thanks for the read